### PR TITLE
feat(cms): past events integration

### PR DESCRIPTION
**Pull request by Kesh-ga**

### CODE REVIEW

File: frontend/pages/event-recaps/[year]/[event_recap_id].js [hunk 1]
```diff
@@ -9,79 +9,52 @@ import Typography from "@material-ui/core/Typography";
...
-      const currEvent = allEvents.filter((event) => {
-        return event.eventNumber.toString() === currEventNumber;
-      })[0];
-      setEvent(currEvent);
-
-      // hide the loading sign
-      setLoading(false);
-
-      setPhotos(currEvent);
+    if (event.fields.resourcesFolderId) {
```
Comment (joanna209) [inline frontend/pages/event-recaps/[year]/[event_recap_id].js:30]: Redirect to 404 page if url is invalid e.g. 2021/400. Currently, the browser throws “application error” when this behaviour occurs

File: frontend/pages/events.js [hunk 2]
```diff
@@ -144,11 +142,12 @@ const Events = () => {
 
   // filter the past events of the given term by the selected category
   const filterTerm = (term, selectedCategory) => {
+    console.log(pastContent[term]);
```
Comment (joanna209) [inline frontend/pages/events.js:145]: Delete console log

File: frontend/pages/events.js [hunk 3]
```diff
@@ -90,13 +91,10 @@ const Events = () => {
...
-      const res = await axios.get(
-        "https://wit-database.herokuapp.com/past-events"
+      const allEvents = allPastEvents.filter(
+        (event) => event.fields.year === Number(year)
       );
-      const allEvents = humps
-        .camelizeKeys(res.data)
-        .filter((event) => event.year === Number(year));
-      setTerms(allEvents.reverse());
+      setTerms(allEvents);
```
Comment (joanna209) [inline frontend/pages/events.js:97]: Reimplement the loading element when user changes the timeline year

File: frontend/lib/api.js [hunk 4]
```diff
@@ -55,3 +55,16 @@ export async function loadPodcasts() {
     });
   return res.items;
 }
+
+export async function loadPastEvents() {
+  const res = await client
+    .getEntries({
+      content_type: "pastEvents",
+      select: "fields",
+      order: "fields.index",
```
Comment (joanna209) [inline frontend/lib/api.js:64]: The index should be reversed

Example: oweek recruitment should be the first event of T1 2022 (ie at the bottom of the past events page) and the cisco event is the last event of T1 2022

**Review by joanna209 — CHANGES_REQUESTED** [review]

File: frontend/pages/events.js [hunk 5]
```diff
@@ -90,13 +91,10 @@ const Events = () => {
     // show loading signs for past events
     setLoadingPast(true);
     const fetchPastEvents = async () => {
-      const res = await axios.get(
-        "https://wit-database.herokuapp.com/past-events"
+      const allEvents = allPastEvents.filter(
```
Comment (joanna209) [inline frontend/pages/events.js:94]: Can we also move the `fetchPastEvents` constant outside of the `useEffect`

**Review by joanna209 — CHANGES_REQUESTED** [review]

File: frontend/pages/events.js (hunk 3 above)
Comment (Kesh-ga) [reply frontend/pages/events.js]: added buffer of 500ms

**Review by joanna209 — APPROVED** [review]

**Pull request merged**